sycl/include/sycl/detail/spirv.hpp

```diff
@@ -501,14 +501,20 @@ AtomicMax(multi_ptr<T, AddressSpace, IsDecorated> MPtr, memory_scope Scope,
 // - The Intel SPIR-V extension natively supports all arithmetic types.
 //   However, OpenCL extension natively supports float vectors,
 //   integer vectors, half scalar and double scalar.
-//   For double vectors we perform emulation with scalar version.
+//   For double, long, long long, unsigned long, unsigned long long
+//   and half vectors we perform emulation with scalar version.
 // - The CUDA shfl intrinsics do not support vectors, and we use the _i32
 //   variants for all scalar types
 #ifndef __NVPTX__
 
 template <typename T>
 struct TypeIsProhibitedForShuffleEmulation
-    : bool_constant<std::is_same_v<vector_element_t<T>, double>> {};
+    : bool_constant<std::is_same_v<vector_element_t<T>, double> ||
```

Comment:
Instead you can create `type_list` of double, long, etc in generic_type_lists.cpp and use `is_contained` here. See examples here:

https://github.com/intel/llvm/blob/c1b2a1c7b07f869ade86a2446954f8ba636bc472/sycl/include/sycl/detail/generic_type_lists.hpp#L92
https://github.com/intel/llvm/blob/c1b2a1c7b07f869ade86a2446954f8ba636bc472/sycl/include/sycl/detail/generic_type_traits.hpp#L51